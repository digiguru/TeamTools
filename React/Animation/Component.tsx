import * as React from "react";

export class BouncyAnimation extends React.Component<any, any> {
     render() {
         const delay       = this.props.delay || "0s";
         const duration    = this.props.duration || "0.8s";
         const toValue     = parseInt(this.props.value || (20), 10);
         const toValueType = this.props.valueType || "%";
         const values      = [
            0 + toValueType,
            (toValue + toValue / 4) + toValueType,
            (toValue - toValue / 10) + toValueType,
            (toValue + toValue / 20) + toValueType,
            (toValue) + toValueType,
         ];
         const valuesToString = values.join(";");
         return <animate attributeType="XML" attributeName={this.props.attributeName} from="0%" to="20%"
                    dur={duration}
                    begin={delay}
                    values={valuesToString}
                    keyTimes="0; 0.3; 0.6; 0.8; 1"
                    fill="freeze"
                    />;
     }
}
