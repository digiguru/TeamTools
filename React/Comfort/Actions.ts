export const ComfortActions = {
    SET_STAGESIZE : "SET_STAGESIZE",
    SET_ZONEFOCUS : "SET_ZONEFOCUS",
    SET_USERFOCUS : "SET_USERFOCUS",
    SELECT_USER   : "SELECT_USER",
    CHOOSE_ZONE   : "CHOOSE_ZONE",
    TOGGLE_CHOICES: "TOGGLE_CHOICES"
};

export function setStageSize(width: number, height: number) {
    return {type: ComfortActions.SET_STAGESIZE, width: width, height: height};
}
export function setUserFocus(user: string, focus: "in-focus" | "active" | "not-in-focus") {
    return {type: ComfortActions.SET_USERFOCUS, user: user, focus: focus};
}
export function setZoneFocus(area: "Chaos" | "Stretch" | "Comfort", focus: "in-focus" | "active" | "not-in-focus") {
    return {type: ComfortActions.SET_ZONEFOCUS, area: area, focus: focus};
}
export function selectUser(user: string) {
    return {type: ComfortActions.SELECT_USER, user: user};
}
export function chooseZone(user: string, area: "Chaos" | "Stretch" | "Comfort", distance: number) {
    return {type: ComfortActions.CHOOSE_ZONE, user: user, area: area, distance: distance};
}
export function toggleChoiceVisibility(visible: boolean) {
    return {type: ComfortActions.TOGGLE_CHOICES, visible: visible};
}
