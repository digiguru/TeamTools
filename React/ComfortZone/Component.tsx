import * as React from "react";
import { IComfortZoneConnectorWithEvents } from "./Model";

export const ReduxChaosArea = (state: IComfortZoneConnectorWithEvents) => { // , onZoneUnfocus, onZoneClick, onZoneActive) => (
    const username = state.User && state.User.Username ? state.User.Username : "Adam Hall";
    return <g>
        <rect id="chaos" className={state.Zone.Focus}
            onMouseEnter={() => state.Events.onZoneOverFocus(state.Zone.Name)}
            onMouseLeave={() => state.Events.onZoneOffFocus(state.Zone.Name)}
            onMouseDown={() => state.Events.onZoneMouseDown(state.Zone.Name)}
            onMouseUp={(event) => state.Events.onZoneMouseUp(username, state.Zone.Name, state.CenterPoint, state.TotalDistance, event)}
            width={state.Zone.Size.Width.toString()} height={state.Zone.Size.Height.toString()}></rect>
        <text className="area-label" id="label-chaos" text-anchor="middle" textAnchor="middle" x="50%" y="20">chaos</text>
    </g>;
};

export const ReduxStretchArea = (state: IComfortZoneConnectorWithEvents) => {
    const username = state.User && state.User.Username ? state.User.Username : "Adam Hall";
    return <g>
            <circle
                className={state.Zone.Focus} id="stretch" r="33%" cx="50%" cy="50%"
                    onMouseEnter={() => state.Events.onZoneOverFocus(state.Zone.Name)}
                    onMouseLeave={() => state.Events.onZoneOffFocus(state.Zone.Name)}
                    onMouseDown={() => state.Events.onZoneMouseDown(state.Zone.Name)}
                    onMouseUp={(event) => state.Events.onZoneMouseUp(username, state.Zone.Name, state.CenterPoint, state.TotalDistance, event)}
                    width={state.Zone.Size.Width.toString()} height={state.Zone.Size.Height.toString()}></circle>
            <text className="area-label" id="label-stretch" text-anchor="middle" textAnchor="middle" x="50%" y="20%">stretch</text>
        </g>;
};

export const ReduxComfortArea = (state: IComfortZoneConnectorWithEvents) => {
    const username = state.User && state.User.Username ? state.User.Username : "Adam Hall";
    return <g>
            <circle
                className={state.Zone.Focus} id="stretch" r="15%" cx="50%" cy="50%"
                    onMouseEnter={() => state.Events.onZoneOverFocus(state.Zone.Name)}
                    onMouseLeave={() => state.Events.onZoneOffFocus(state.Zone.Name)}
                    onMouseDown={() => state.Events.onZoneMouseDown(state.Zone.Name)}
                    onMouseUp={(event) => state.Events.onZoneMouseUp(username, state.Zone.Name, state.CenterPoint, state.TotalDistance, event)}
                    width={state.Zone.Size.Width.toString()} height={state.Zone.Size.Height.toString()}></circle>
            <text className="area-label" id="label-stretch" text-anchor="middle" textAnchor="middle" x="50%" y="50%">comfort</text>
        </g>;
};
