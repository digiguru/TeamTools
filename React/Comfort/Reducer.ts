
import {ComfortActions} from "./Actions";
import {ComfortAppState} from "./Model";
import { fromJS, List, Map } from "immutable";
import { Point } from "../Models/Point";
import { Size } from "../Models/Size";
import { DOMMeasurement } from "../Models/IDomMeasurement";


const initialSize: Size = new Size(800, 800);
const initialState: ComfortAppState = {
    Size: initialSize,
    CenterPoint: new Point(initialSize.width / 2, initialSize.height / 2),
    UserList : {
        ShowUsers: true,
        Users: [
            {Username: "Adam Hall", Focus: "not-in-focus", Y: 0 },
            {Username: "Caroline Hall", Focus: "not-in-focus", Y: 0 }
        ]
    },
    Zones : {
        Comfort: {Name: "Comfort", Focus: "not-in-focus", Range: {Start: 0, End: 100}, Size: {Width: new DOMMeasurement("50%"), Height: new DOMMeasurement("50%")}},
        Stretch: {Name: "Stretch", Focus: "not-in-focus", Range: {Start: 100, End: 200}, Size: {Width: new DOMMeasurement("50%"), Height: new DOMMeasurement("50%")}},
        Chaos: {Name: "Chaos", Focus: "not-in-focus", Range: {Start: 200, End: 300}, Size: {Width: new DOMMeasurement("100%"), Height: new DOMMeasurement("100%")}}
    },
    ShowUserChoices: false,
    UserChoices: []
};
export function comfortReactApp(state: ComfortAppState = initialState, action): ComfortAppState {
    console.log(action.type, action);
    switch (action.type) {
        case ComfortActions.SET_STAGESIZE:
            return ComfortZoneAction.setStageSize(state, (<any>action).width, (<any>action).height);
        case ComfortActions.SET_USERFOCUS:
            return ComfortZoneAction.setUserFocus(state, (<any>action).user, (<any>action).focus);
        case ComfortActions.SET_ZONEFOCUS:
            return ComfortZoneAction.setZoneFocus(state, (<any>action).area, (<any>action).focus);
        case ComfortActions.SELECT_USER:
            return ComfortZoneAction.selectUser(state, (<any>action).user);
        case ComfortActions.CHOOSE_ZONE:
            return ComfortZoneAction.chooseZone(state, (<any>action).user, (<any>action).area, (<any>action).distance, (<any>action).x, (<any>action).y);
        case ComfortActions.TOGGLE_CHOICES:
            return ComfortZoneAction.toggleChoiceVisibility(state, (<any>action).visible);
        default:
            return state;
    }
}
class ComfortZoneAction {
    static setStageSize(state: ComfortAppState, width: number, height: number) {
        const newCenter = new Point(width / 2, height / 2);
        return fromJS(state)
            .set("Size", new Size(width, height))
            .set("CenterPoint", newCenter)
            .toJS();
    }
    static setZoneFocus(state: ComfortAppState, area: "Chaos" | "Stretch" | "Comfort", focus: "in-focus" | "active" | "not-in-focus"): ComfortAppState {
        return fromJS(state)
            .setIn(["Zones", "Comfort", "Focus"], area === "Comfort" ? focus : "not-in-focus")
            .setIn(["Zones", "Stretch", "Focus"], area === "Stretch"  ? focus : "not-in-focus")
            .setIn(["Zones", "Chaos", "Focus"], area === "Chaos"  ? focus : "not-in-focus").toJS();
    }
    static setUserFocus(state: ComfortAppState, user: string, focus: "in-focus" | "active" | "not-in-focus"): ComfortAppState {
        const originalList = List(state.UserList.Users);
        const newUserList = originalList.update(
            originalList.findIndex(item => item.Username === user),
            (item) => fromJS(item).set("Focus", focus)).toJS();
        return fromJS(state)
            .setIn(["UserList", "Users"], newUserList).toJS();
    }
    static selectUser(state: ComfortAppState, user: String): ComfortAppState {
        // Sets currentUser, and therefor hides the user choice menu
        const data = fromJS(state)
            .set("CurrentUser", user)
            .set("ShowUserChoices", false)
            .setIn(["UserList", "ShowUsers"], false);

        return data.toJS();
    }

    static chooseZone(state: ComfortAppState, user: string, area: "Chaos" | "Stretch" | "Comfort", distance: number, x: number, y:number): ComfortAppState {

        // Add the user choice
        const newUserChoices = List(state.UserChoices).push({
            User: user,
            Zone: area,
            Distance: distance
        }).toJS();
        // Remove the user from the choice list
        const newUserList = List(state.UserList.Users).filter((item) => item.Username !== user).toArray();
        // Show the user list
        const showUserChoice = !!(newUserList.length);
        // Return
        return fromJS(state)
            .delete("CurrentUser")
            .set("CenterPoint", new Point(x,y))
            .set("ShowUserChoices", showUserChoice)
            .set("UserChoices", newUserChoices)
            .setIn(["UserList", "Users"], newUserList)
            .setIn(["UserList", "ShowUsers"], showUserChoice).toJS();
    }
    static toggleChoiceVisibility(state: ComfortAppState, visible: boolean): ComfortAppState {
        // Set "showUserChoices" to true
        return Map(state)
            .set("ShowUserChoices", visible).toJS();
    };
}