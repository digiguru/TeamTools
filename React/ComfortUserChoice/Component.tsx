import * as React from "react";
import {ChaosPickerUserChoiceState} from "../Comfort/Model";
import {Point} from "../Models/Point";
import {Polar} from "../Models/Polar";
import { IComfortUserChoice, IUserChoiceState } from "./Model";

export const ReduxUserHistoryArea = (state: IUserChoiceState) => {
    if (state && state.Choices.length) {
        const totalPoints = state.Choices.length;
        const radian = 6.2831853072; // 360 * Math.PI / 180;
        const polarDivision = radian / totalPoints;
        return <g id="history">
            {state.Choices.map((userChoice: ChaosPickerUserChoiceState, i) =>
                <ReduxUserHistory CenterPoint={state.CenterPoint} key={userChoice.User.Username} {...userChoice} Index={i} PolarDivision={polarDivision}  />
            )}
        </g>;

    } else {
        return <g id="history"></g>;
    }
};

export const ReduxUserHistory = (state: IComfortUserChoice) => {
    const angle = state.PolarDivision * state.Index;
    const point = Point.toCartesian(new Polar(state.Distance, angle), state.CenterPoint);
    return <circle cx={point.x} cy={point.y} r="10" className="point"></circle>;
};