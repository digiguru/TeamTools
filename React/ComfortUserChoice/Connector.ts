import { connect } from "react-redux";
import { ReduxUserHistoryArea} from "./Component";
import { ComfortAppState } from "../Comfort/Model";
import { IUserChoiceState } from "./Model";

const mapStateToProps = (state: ComfortAppState, ownProps: IUserChoiceState): IUserChoiceState => {
    return {
      Choices    : state.UserChoices,
      CenterPoint: state.CenterPoint,
      MaxDistance: state.Size.shortest()
    };
};


const mapDispatchToProps = (dispatch) => {
  return {};
};

export const ReduxUserHistoryConnector = connect(
  mapStateToProps,
  mapDispatchToProps
)(ReduxUserHistoryArea);

// UserListConnector
