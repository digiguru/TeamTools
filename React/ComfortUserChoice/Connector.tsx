import { connect } from "react-redux";
import {ReduxUserHistory, ReduxUserHistoryArea} from "./Component";
import {ComfortAppState, ChaosPickerUserChoiceState} from "../Comfort/Model";
import { fromJS } from "immutable";
import { Point } from "../Models/Point";

export interface IUserChoiceState {
  Choices: Array<ChaosPickerUserChoiceState>;
  CenterPoint: Point;
}

const mapStateToProps = (state: ComfortAppState, ownProps: IUserChoiceState): IUserChoiceState => {
    return {
      Choices: state.UserChoices,
      CenterPoint: state.CenterPoint
    };
};


const mapDispatchToProps = (dispatch) => {
  return {};
};

export const ReduxUserHistoryConnector = connect(
  mapStateToProps,
  mapDispatchToProps
)(ReduxUserHistoryArea);

// UserListConnector
