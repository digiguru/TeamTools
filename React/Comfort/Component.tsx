import * as React from "react";
import {ReduxChaosConnector, ReduxStretchConnector, ReduxComfortConnector} from "../ComfortZone/Connector";
import {ReduxUserConnector} from "../User/Connector";
import {ReduxUserHistoryConnector} from "../ComfortUserChoice/Connector";
import { StageConnector } from "../Stage/Connector";

export const ComfortStage = (state) => (
   <g>
        <g>
            <ReduxChaosConnector   Name="Chaos" />
            <ReduxStretchConnector Name="Stretch" />
            <ReduxComfortConnector Name="Comfort" />
            <ReduxUserConnector />
            <ReduxUserHistoryConnector />
        </g>,
        <button id="show-tuckman" onMouseUp={state.onShow}>show tuckman</button>,
        <button id="hide-tuckman" onMouseUp={state.onHide}>hide tuckman</button>
    </g>
);
