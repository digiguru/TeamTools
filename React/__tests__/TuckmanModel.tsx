import * as React from "react";
import { createStore } from "../../3rdParty/redux.min";
import { Provider } from "../../3rdParty/react-redux.min";
import { TuckmanApp, TuckmanStage } from "../Tuckman/Component";
import { tuckmanReactApp } from "../Tuckman/Reducer";
import { Size } from "../Models/Size";
import * as Action from "../Tuckman/Actions";
import { TuckmanConnector } from "../Tuckman/Connector";
import { StageConnector } from "../Stage/Connector";
import { Stage } from "../Stage/Component";
const renderizer = require("react-test-renderer");

test.skip("Should not mutate in any way", () => {
    const myStore = createStore(tuckmanReactApp);
    const originalState = myStore.getState();
    const inputState = JSON.stringify(originalState);
    const checkAfterAction = (action) => {
        myStore.dispatch(action);
        expect(inputState).toEqual(JSON.stringify(originalState));
    };
    checkAfterAction(Action.setUserFocus("Adam Hall", "in-focus"));
    checkAfterAction(Action.selectUser("Adam Hall"));
    checkAfterAction(Action.setStageSize(800, 600));
    checkAfterAction(Action.setZoneFocus("forming", "in-focus"));
    checkAfterAction(Action.setZoneFocus("storming", "active"));
    checkAfterAction(Action.setZoneFocus("norming", "in-focus"));
    checkAfterAction(Action.setZoneFocus("performing", "in-focus"));
    checkAfterAction(Action.toggleChoiceVisibility(true));
    checkAfterAction(Action.toggleChoiceVisibility(false));
    checkAfterAction(Action.chooseZone("Adam Hall", "performing", 85));
});

test.skip("Focusable zones", () => {
    debugger;
    const myStore = createStore(tuckmanReactApp);
    myStore.dispatch(Action.setStageSize(800, 600));
    const size = new Size(800, 600);
    const component = renderizer.create(

            <Provider store={myStore}>
                <StageConnector>
                    <TuckmanConnector />
                </StageConnector>
            </Provider>

    );
    console.log(myStore.getState());

    myStore.dispatch(Action.setZoneFocus("storming", "in-focus"));
    console.log(myStore.getState());

    expect(component.toJSON()).toMatchSnapshot();
    expect(myStore.getState()).toMatchSnapshot();
});

test("Should show the component", () => {
    // Arrange
    debugger;
    const myStore = createStore(tuckmanReactApp);
    const stageSize = new Size(800, 600);
    myStore.dispatch(Action.setStageSize(800, 600));
    const component = renderizer.create(
            <Provider store={myStore}>
                <StageConnector>
                    <TuckmanConnector />
                </StageConnector>
            </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
    myStore.dispatch(Action.setZoneFocus("forming", "in-focus"));
    expect(component.toJSON()).toMatchSnapshot();

});
/*
it("Should show the stretch area", () => {
    // Arrange
    const component = renderizer.create(
        <Stage><ChartArea width="200" offset="100" label="example"></ChartArea></Stage>
    );
    expect(component.toJSON()).toMatchSnapshot();

});


*/
