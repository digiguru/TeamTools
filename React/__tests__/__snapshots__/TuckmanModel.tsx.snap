exports[`test Should show the component 1`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <svg
    height="800"
    id="stage"
    width="800">
    <g
      id="zones">
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="performing"
          onMouseDown={undefined}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onMouseUp={undefined}
          width="25%"
          x="0"
          y="0">
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.6000000000000001s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;93.75%;67.5%;78.75%;75%" />
        </rect>
        <text
          className="area-label"
          id="performing-label"
          text-anchor="middle"
          textAnchor="middle"
          x="87%"
          y="50%">
          performing
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="norming"
          onMouseDown={undefined}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onMouseUp={undefined}
          width="25%"
          x="0"
          y="0">
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.4s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;62.5%;45%;52.5%;50%" />
        </rect>
        <text
          className="area-label"
          id="norming-label"
          text-anchor="middle"
          textAnchor="middle"
          x="62%"
          y="50%">
          norming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="storming"
          onMouseDown={undefined}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onMouseUp={undefined}
          width="25%"
          x="0"
          y="0">
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.2s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;31.25%;22.5%;26.25%;25%" />
        </rect>
        <text
          className="area-label"
          id="storming-label"
          text-anchor="middle"
          textAnchor="middle"
          x="37%"
          y="50%">
          storming
        </text>
        ;
      </g>
      <g>
        <rect
          className="not-in-focus area okay js-area-standard"
          height="100%"
          id="forming"
          onMouseDown={undefined}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onMouseUp={undefined}
          width="25%"
          x="0"
          y="0">
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;0%;0%;0%;0%" />
        </rect>
        <text
          className="area-label"
          id="forming-label"
          text-anchor="middle"
          textAnchor="middle"
          x="12%"
          y="50%">
          forming
        </text>
        ;
      </g>
    </g>
  </svg>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <svg
    height="800"
    id="stage"
    width="800"
  >
    <g id="zones">
      <g>
        <rect
          class="not-in-focus area okay js-area-standard"
          height="100%"
          id="performing"
          width="25%"
          x="0"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.6000000000000001s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;93.75%;67.5%;78.75%;75%"
          >
          </animate>
        </rect>
        <text
          class="area-label"
          id="performing-label"
          text-anchor="middle"
          textAnchor="middle"
          x="87%"
          y="50%"
        >
          performing
        </text>
        ;
      </g>
      <g>
        <rect
          class="not-in-focus area okay js-area-standard"
          height="100%"
          id="norming"
          width="25%"
          x="0"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.4s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;62.5%;45%;52.5%;50%"
          >
          </animate>
        </rect>
        <text
          class="area-label"
          id="norming-label"
          text-anchor="middle"
          textAnchor="middle"
          x="62%"
          y="50%"
        >
          norming
        </text>
        ;
      </g>
      <g>
        <rect
          class="not-in-focus area okay js-area-standard"
          height="100%"
          id="storming"
          width="25%"
          x="0"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0.2s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;31.25%;22.5%;26.25%;25%"
          >
          </animate>
        </rect>
        <text
          class="area-label"
          id="storming-label"
          text-anchor="middle"
          textAnchor="middle"
          x="37%"
          y="50%"
        >
          storming
        </text>
        ;
      </g>
      <g>
        <rect
          class="not-in-focus area okay js-area-standard"
          height="100%"
          id="forming"
          width="25%"
          x="0"
          y="0"
        >
          <animate
            attributeName="x"
            attributeType="XML"
            begin="0s"
            dur="0.8s"
            fill="freeze"
            from="0%"
            keyTimes="0; 0.3; 0.6; 0.8; 1"
            to="20%"
            values="0%;0%;0%;0%;0%"
          >
          </animate>
        </rect>
        <text
          class="area-label"
          id="forming-label"
          text-anchor="middle"
          textAnchor="middle"
          x="12%"
          y="50%"
        >
          forming
        </text>
        ;
      </g>
    </g>
  </svg>
</svg>
`;

exports[`test Should show the stretch area 1`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <rect
      className="not-in-focus area okay js-area-standard"
      height="100%"
      id="example"
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      width="25%"
      x="0"
      y="0">
      <animate
        attributeName="x"
        attributeType="XML"
        begin="0s"
        dur="0.8s"
        fill="freeze"
        from="0%"
        keyTimes="0; 0.3; 0.6; 0.8; 1"
        to="20%"
        values="0%;0%;0%;0%;0%" />
    </rect>
    <text
      className="area-label"
      id="example-label"
      text-anchor="middle"
      textAnchor="middle"
      x="12%"
      y="50%">
      example
    </text>
    ;
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <rect
      class="not-in-focus area okay js-area-standard"
      height="100%"
      id="example"
      width="25%"
      x="0"
      y="0"
    >
      <animate
        attributeName="x"
        attributeType="XML"
        begin="0s"
        dur="0.8s"
        fill="freeze"
        from="0%"
        keyTimes="0; 0.3; 0.6; 0.8; 1"
        to="20%"
        values="0%;0%;0%;0%;0%"
      >
      </animate>
    </rect>
    <text
      class="area-label"
      id="example-label"
      text-anchor="middle"
      textAnchor="middle"
      x="12%"
      y="50%"
    >
      example
    </text>
    ;
  </g>
</svg>
`;
